irods/manager/user_manager.py

```diff
@@ -187,25 +202,42 @@ def get(self, name, user_zone=""):
             raise UserGroupDoesNotExist()
         return iRODSUserGroup(self, result)
 
-    def create(self, name, user_type='rodsgroup', user_zone="", auth_str="", group_admin=False):
-        if group_admin:
-            message_body = UserAdminRequest(
+    def _api_info(self, group_admin_flag):
+        """
+        If group_admin_flag is:   then use UserAdminRequest API:
+        --------------------      ------------------------------
+        True                      always
+        False                     never
+        None                      when user type is groupadmin
+        """
+
+        sess = self.sess
+        if (sess.users.get(sess.username).type == 'groupadmin' and group_admin_flag is not False) or group_admin_flag:
```

Comment:
Does retrieval of the user type require a network call? If so, consider reordering the checks to avoid that.

Reply:
reordered for overall network efficiency.